src/coreclr/jit/gentree.h

```diff
@@ -562,6 +562,12 @@ enum GenTreeFlags : unsigned int
     GTF_OVERFLOW                = 0x10000000, // Supported for: GT_ADD, GT_SUB, GT_MUL and GT_CAST.
                                               // Requires an overflow check. Use gtOverflow(Ex)() to check this flag.
 
+#ifdef TARGET_ARM64
+    GTF_DIV_BY_ZERO_CHK         = 0x20000000, // GT_DIV -- Tells ARM64 code-gen to possibly emit a divide-by-zero check.
+
+    GTF_DIV_OVERFLOW_CHK        = 0x40000000, // GT_DIV -- Tells ARM64 code-gen to possibly emit an overflow check.
+#endif // TARGET_ARM64
```

Comment:
We should follow the existing patterns with these flags. The existing pattern here is `GTF_IND_NONFAULTING`.

The meaning of the flags should such that a tree without them is conservatively correct (may throw).

Then, `OperExceptions` should be taught to look at these flags when determining whether a node is throwing and the flag be made all-platform.

Reply:
I've thought about this too, but am a little afraid of what regressions may occur. Wondering if this should also be a separate PR.

Reply:
Why would there be regressions? The "before" state is that the division is always throwing unless directly provable otherwise, in the "after" state there would simply be more cases where it is known to not throw (via the flags).

That aside, the meaning of the flags needs to be flipped; there is no reason to introduce fragility into the system.

Reply:
So you want to have a GTF_DIV_NO_DIV_BY_ZERO_CHK?

Reply:
Yes.

Reply:
Gotcha, makes more sense now. I actually tried this before but ended up getting less improvements, so I can try it again now that everything appears to be working.